ui/src/components/drafts/components/DraftEditorHeader.js

```diff
@@ -39,51 +41,96 @@ class DraftEditorHeader extends React.Component {
     this.state = { actionType: null };
   }
 
-  _validateFormData() {
+  _validateFormData = (form_type = "form", show_toast = false) => {
     // TOFIX maybe fetch formData from store instead of ref
-    const formData = this.props.formRef.current.props.formData;
+    const formData = this.props.formRef.current
+      ? this.props.formRef.current.props.formData
+      : null;
     const { errors } = this.props.formRef.current.validate(formData);
 
-    let e = new Event("save");
+    this.props.formRef.current.submit();
+
+    // let undefinedNumber = 0;
+    // let sum = Object.keys(formData).length;
+    // Object.keys(formData).forEach(
+    //   key => formData[key] === undefined && undefinedNumber++
+    // );
+
+    let emptyObject = 0;
+
+    for (let [key, value] of Object.entries(formData)) {
+      if (_isEmpty(value)) {
+        emptyObject++;
+      } else {
+        for (let [innerKey, innerValue] of Object.entries(value)) {
+          if (_isEmpty(innerValue)) {
+            emptyObject++;
+          }
+        }
+      }
+    }
 
-    this.props.formRef.current.onSubmit(e);
+    let condition =
+      _isEmpty(formData) || emptyObject === Object.entries(formData).length;
+    // undefinedNumber === sum ||
+    // Object.entries(formData).length === 0;
 
     if (errors.length > 0) {
+      cogoToast.error("Make sure all the fields are properly filled in", {
+        position: "top-center",
+        heading: "Form could not be submitted",
+        bar: { size: "0" },
+        hideAfter: 5
+      });
       return false;
-    } else {
-      return true;
     }
-  }
 
-  _createDraft(schema_id) {
-    if (
-      this.props.formData == null ||
-      this.formData == {} ||
-      this.formData == ""
-    ) {
-      this.props.createDraftError(
-        "Form is empty. Please add some content first and try saving again"
+    if (condition) {
+      cogoToast.warn(
+        "Please add some content first, and try again saving again",
+        {
+          position: "top-center",
+          heading: "Form is empty",
+          bar: { size: "0" },
+          hideAfter: 3
+        }
       );
-      return;
+      return false;
+    } else {
+      // for the success message you can define whether you want to display it or not
+      show_toast
```

Comment:
does it take into account server response?

Reply:
Since there are two calls for validation, and in the case of success it will go through the toast twice it will pop up the message twice. Therefore I pass a parameter at the second call to show the toast, by default the message will not be displayed